src/main/java/ee/carlrobert/llm/client/ollama/OllamaClient.java

```diff
@@ -0,0 +1,255 @@
+package ee.carlrobert.llm.client.ollama;
+
+import static java.lang.String.format;
+
+import com.fasterxml.jackson.core.JacksonException;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import ee.carlrobert.llm.PropertiesLoader;
+import ee.carlrobert.llm.client.DeserializationUtil;
+import ee.carlrobert.llm.client.StreamableRequest;
+import ee.carlrobert.llm.client.ollama.completion.request.OllamaCompletionRequest;
+import ee.carlrobert.llm.client.ollama.completion.request.OllamaEmbeddingRequest;
+import ee.carlrobert.llm.client.ollama.completion.request.OllamaPullRequest;
+import ee.carlrobert.llm.client.ollama.completion.response.OllamaCompletionResponse;
+import ee.carlrobert.llm.client.ollama.completion.response.OllamaEmbeddingResponse;
+import ee.carlrobert.llm.client.ollama.completion.response.OllamaModelInfoResponse;
+import ee.carlrobert.llm.client.ollama.completion.response.OllamaPullResponse;
+import ee.carlrobert.llm.client.ollama.completion.response.OllamaTagsResponse;
+import ee.carlrobert.llm.client.openai.completion.ErrorDetails;
+import ee.carlrobert.llm.completion.CompletionEventListener;
+import ee.carlrobert.llm.completion.CompletionEventSourceListener;
+import java.io.IOException;
+import java.util.Map;
+import okhttp3.Interceptor;
+import okhttp3.MediaType;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.RequestBody;
+import okhttp3.Response;
+import okhttp3.ResponseBody;
+import okhttp3.sse.EventSource;
+import okhttp3.sse.EventSources;
+import org.jetbrains.annotations.NotNull;
+
+public class OllamaClient {
+
+  private static final String BASE_URL = PropertiesLoader.getValue("ollama.baseUrl");
+
+  private final OkHttpClient httpClient;
+  private final String host;
+  private final Integer port;
+
+  protected OllamaClient(Builder builder, OkHttpClient.Builder httpClientBuilder) {
+    httpClientBuilder
+        .addInterceptor(new Interceptor() {
+          @NotNull
+          @Override
+          public Response intercept(@NotNull Chain chain) throws IOException {
+            Request request = chain.request();
+            Response response = chain.proceed(request);
+            // Workaround since OkHttp does not recognize "application/x-ndjson" as a valid stream
+            // Content-Type
+            // see: RealEventSource.kt ResponseBody.isEventStream()
+            if (response.header("Content-Type", "")
```

Comment:
We have a problem with the `OkHttp` Client, since it uses the `text/event-stream` Content-Type, but Ollama returns `application/x-ndjson`. I did not see any other way than using an `Interceptor` to rewrite Ollama's response to a `text/event-stream`, but there is probably a better way.

Reply:
I think we should add the possibility to the Response headers and such too, which is not yet possible with the `expectOllama(StreamHttpExchange)`

Reply:
Yeah, I don't know any better way right now, either.